src/components/InputProblem.tsx

```diff
@@ -1,7 +1,92 @@
+import React, { useState, useCallback } from 'react'
 import { BaseProblemProps } from './ProblemSetBlock'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import { mathifyElement } from '../lib/math'
+import * as Yup from 'yup'
 
-interface InputProblemProps extends BaseProblemProps {}
+interface InputProblemProps extends BaseProblemProps {
+  comparator: string
+}
+
+interface InputSchema {
+  response: string | number
+}
+
+interface InputFormValues {
+  response: string
+}
+
+export const InputProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback,
+  solution, retryLimit, content, comparator, encourageResponse, buttonText, correctResponse
+}: InputProblemProps): JSX.Element => {
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const [inputDisabled, setInputDisabled] = useState(false)
+  const [feedback, setFeedback] = useState('')
+
+  const schema = (): Yup.SchemaOf<InputSchema> => {
+    if (comparator === 'integer') {
+      return Yup.object({
+        response: Yup.number().integer().typeError('Please provide an Integer').required('Please provide an Integer')
+      })
+    }
+    if (comparator === 'float') {
+      return Yup.object({
+        response: Yup.number().typeError('Please provide an number').required('Please provide a number')
+      })
+    }
+    return Yup.object({
+      response: Yup.string().trim().required('Please provide valid input')
+    })
+  }
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const handleSubmit = async (values: InputFormValues): Promise<void> => {
+    if (values.response.trim() === solution.trim()) {
+      setFeedback(correctResponse)
+      solvedCallback()
+      setInputDisabled(true)
+    } else if (retryLimit === 0 || retriesAllowed !== retryLimit) {
+      setRetriesAllowed(currRetries => currRetries + 1)
+      setFeedback(encourageResponse)
+      allowedRetryCallback()
+    } else {
+      exhaustedCallback()
+      setFeedback('No more attempts allowed')
```

Comment:
Can you incorporate use of [this shared constant](https://github.com/openstax/k12-apps-raise/blob/6c7c3d02a0e25fb0e6800800c87c737389a5d300/src/components/ProblemSetBlock.tsx#L31) for the string here?